### PR TITLE
Feature/#320 log page

### LINKED ISSUE #320: Log page for app / web instance
* Log page for app / web instance

## Requirements
 * [ ] Create log stash for local logs
 * [ ] Create instance for logs
 * [ ] Paginate logs

### PR DESCRIPTION
**Pull request by inureyes**
This PR resolves #320

### CODE REVIEW

File: src/components/backend-ai-error-log-list.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+          </vaadin-grid>
+        `;
+      }
+ }
+
+ declare global {
+    interface HTMLElementTagNameMap {
+      "backend-ai-error-log-list": BackendAiErrorLogList;
+    }
+ }
```
Comment (adrysn) [inline src/components/backend-ai-error-log-list.ts]: This file has one space before every line. Please remove them if this is not intended. Also, the default indentation is two spaces, not three or four.

File: src/components/backend-ai-error-log-view.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        </wl-card>
+        `;
+    }
+ }
+
+ declare global {
+     interface HTMLElementTagNameMap {
+         "backend-ai-error-log-view": BackendAIErrorLogView;
+     }
+ }
```
Comment (adrysn) [inline src/components/backend-ai-error-log-view.ts]: This file also has one space before every line.

File: src/components/backend-ai-usersettings-view.ts [hunk 3]
```diff
@@ -184,13 +257,27 @@ export default class BackendAiUserSettingsView extends BackendAIPage {
     } else { // already connected
       this.updateSettings();
     }
-    this.indicator = this.shadowRoot.querySelector('#loading-indicator');
+    this.indicator = this.shadowRoot.querySelector(
+      !'#loading-indicator');
```
Comment (adrysn) [inline src/components/backend-ai-usersettings-view.ts]: Is this a typo?

File: src/components/backend-ai-usersettings-view.ts [hunk 4]
```diff
@@ -236,16 +323,53 @@ export default class BackendAiUserSettingsView extends BackendAIPage {
   }
 
   async _editBootstrapScript() {
-    const editor = this.shadowRoot.querySelector('#codemirror-editor');
-    const dialog = this.shadowRoot.querySelector('#bootstrap-dialog');
     const script = await this._fetchBootstrapScript();
-    editor.setValue(script);
-    dialog.show();
+    this.bootstrapDialog.setValue(script);
```
Comment (adrysn) [inline src/components/backend-ai-usersettings-view.ts:274]: Why is it changed from `#codemirror-editor` to `#bootstrap-dialog`? I think bootstrap dialog does not have `setValue` method?

**Comment by adrysn** [comment]
In user settings page, `Preferences` and `Logs / Errors` menu in the upper right corner does not work. This may be due to the fact that url does not change when calling `window.history.pushState`... 

image

File: src/components/backend-ai-usersettings-view.ts [hunk 5]
```diff
@@ -236,16 +323,53 @@ export default class BackendAiUserSettingsView extends BackendAIPage {
...
+    if (currentLogs) {
+    localStorage.removeItem('backendaiconsole.logs');
+    }
+    let event = new CustomEvent("log-message-clear", {});
+    document.dispatchEvent(event);
+    localStorage.getItem('backendaiconsole.logs');
+    this.clearLogsDialog.hide();
+    this.notification.text = 'Log Messages have been removed.';
+    this.notification.show();
+    this.indicator.hide();
```
Comment (adrysn) [inline src/components/backend-ai-usersettings-view.ts:297]: After clearing logs, client-side error occurs.
스크린샷 2020-02-12 오전 1 25 54

**Comment by adrysn** [comment]
Group(project) list is ruined for admins.

image

This superadmin is only belong to one group, h-group-A-a:

image

**Comment by inureyes** [comment]
LGTM. I'll merge this.

**Pull request merged**